src/corell/src/queue.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 The Gfx-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Command queue types.
+//!
+//! There are different types of queues, which can create and submit associated command buffers.
+
+use std::ops::{Deref, DerefMut};
+use {CommandQueue};
+use command::Submit;
+
+/// General command queue, which can execute graphics, compute and transfer command buffers.
+pub struct GeneralQueue<Q: CommandQueue>(Q);
+impl<Q: CommandQueue> GeneralQueue<Q> {
+    #[doc(hidden)]
+    pub unsafe fn new(queue: Q) -> Self {
+        GeneralQueue(queue)
+    }
+
+    pub fn submit_general(&mut self, cmd_buffers: &[Submit<<Q as CommandQueue>::GeneralCommandBuffer>]) {
```

Comment:
I don't think you need to write `<Q as CommandQueue>` here and below, can just use `Q`